anndata/_io/specs/registry.py

```diff
@@ -101,9 +235,17 @@ def _register(func):
 
         return _register
 
-    def get_partial_reader(self, src_type, spec, modifiers=frozenset()):
+    def get_partial_reader(
+        self,
+        src_type: type,
+        spec: IOSpec | Mapping[str, str],
+        modifiers: Iterable[str] = frozenset(),
+    ):
         modifiers = frozenset(modifiers)
-        return self.read_partial[(src_type, spec, modifiers)]
+        try:
+            return self.read_partial[(src_type, spec, modifiers)]
+        except KeyError:
+            raise NoSuchRead("read_partial", (src_type, spec, modifiers)) from None
```

Comment:
Similar to above, would prefer `if` over `try`

Reply:
that’s the pythonic way:

- https://docs.python.org/3.7/glossary.html#term-eafp
- https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/